.github/workflows/codspeed.yml

```diff
@@ -0,0 +1,78 @@
+name: CodSpeed
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - 'justfile'
+      - '.devcontainer'
+      - '**/*.md'
+      - '.gitignore'
+  merge_group:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: codspeed.yml-${{ github.event.pull_request.number || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  # Write cache
+  contents: write
+
+jobs:
+  benchmarks:
+    name: Run benchmarks
+    runs-on: ubuntu-22.04
+    env:
+      BAZEL_ARGS: --config=benchmark --@codspeed//core:codspeed_mode=instrumentation --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          show-progress: false
+
+      - name: Cache
+        id: cache
+        uses: actions/cache@v4
+        with:
+          path: ~/bazel-disk-cache
+          key: bazel-disk-cache-benchmarks-${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE') }}
+
+      - name: Setup Linux
+        run: |
+          export DEBIAN_FRONTEND=noninteractive
+          wget https://apt.llvm.org/llvm.sh
+          sed -i '/apt-get install/d' llvm.sh
+          chmod +x llvm.sh
+          sudo ./llvm.sh 19
+          # keep in sync with build/ci.bazelrc
+          sudo apt-get install -y -qq --no-install-recommends clang-19 lld-19 libunwind-19 libc++abi1-19 libc++1-19 libc++-19-dev libclang-rt-19-dev llvm-19
+          sudo ln -s /usr/bin/llvm-symbolizer-19 /usr/bin/llvm-symbolizer
+          echo "build:linux --action_env=CC=/usr/lib/llvm-19/bin/clang" >> .bazelrc
+          echo "build:linux --host_action_env=CC=/usr/lib/llvm-19/bin/clang" >> .bazelrc
+
+      - name: Build benchmarks
+        run: |
+          bazel build ${{ env.BAZEL_ARGS }} //src/workerd/tests:all_benchmarks
```

Comment:
This means we need to build another workerd binary for benchmarks, we should add the benchmarks as an extra step on top of one of our already running jobs using its artifacts rather than another full build.
There's existing examples in test.yml for generated types and wpt reports etc.

Reply:
If we want to reduce the duration, we should add a large runner as well: https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners/managing-larger-runners#adding-a-larger-runner-to-an-organization

@danlapid I believe you are the only one who can add this.

Reply:
The benchmark flag is not really changing the compile flags, does it?
It just enables compilation of a few extra files?

Reply:
Ideally it changes compile flags. It should run proper optimization flags etc.